### DISCUSSION TITLE
Use Digital Ocean deploy production standard operating procedures

### DISCUSSION THREAD

**Discussion by fire** [Budibase/budibase#2752] [Help]
Hi!

Is there documentation for enabling the Digital Ocean deploy with encrypted connections?

I also wanted to enabled Digital Ocean spaces (S3 Compatible storage).

Alternatively I can use K8S deploy with Kubernetes, but I can't find any standard procedure for helm with encrypted connection and DO spaces (S3).

How do I backup the entire instance? The user based backup only does non-attachments.

**Comment by mjashanks** [comment]
HI @fire 

Here are some docs on setting up a reverse proxy with Budibase - https://docs.budibase.com/self-hosting/advanced-options/reverse-proxy#nginx-advanced

You can set this up directly on your DO droplet or on a separate droplet.

For file serving and attachment, Budibase uses MinIO - which is one of the docker containers included. Minio is S3 compatible, so it should be possible to get Budibase to point to another S3 compatible service. However, this is undocumented, so you'd need to figure it out for your own setup. You will need to fiddle with your https://github.com/Budibase/budibase/blob/develop/hosting/envoy.yaml - which proxys requests to MinIO.

To backup, you will need to backup both MinIO and CouchDB. This discussion: https://github.com/Budibase/budibase/discussions/1657 - talks about how to access the CouchDB UI. Minio can be access on http://<domain>/minio/login.
